mesa/datacollection.py

```diff
@@ -80,14 +80,17 @@ def __init__(self, model_reporters={}, agent_reporters={}, tables={}):
         self.agent_vars = {}
         self.tables = {}
 
-        for name, func in model_reporters.items():
-            self._new_model_reporter(name, func)
+        if isinstance(model_reporters, dict):
```

Comment:
Would it be faster or more idiomatic with `if model_reporters is not None`?

Reply:
TL;DR - I'll change it.

Long version - In this case, since the expected type is of type `dict`, it makes sense that we only want objects that are of that type. Though I suppose you'll just get errors if you pass in something that doesn't have a `__getitem__`  method. In think about this more, you're probably right that it should just be a check against `None`, and then the user can dig through the stack trace to discover that the object that they're wanting to use is a dictionary of some kind.

Reply:
Yeah, with `isinstance(model_reporters, dict)`, the error doesn't show up soon when it is not a dict.